pkg/reconciler/finalizer/reposync_finalizer.go

```diff
@@ -29,18 +29,18 @@ import (
 
 // RepoSyncFinalizer handles finalizing RepoSync objects, using the Destroyer
 // to destroy all managed user objects previously applied from source.
-// Impliments the Finalizer interface.
+// Implements the Finalizer interface.
 type RepoSyncFinalizer struct {
 	Destroyer applier.Destroyer
 	Client    client.Client
 
-	// StopControllers will be called by the Finalize() method to stop the Parser and Remediator.
+	// StopControllers will be called by the Finalize() method to stop the Syncer and Remediator.
 	StopControllers context.CancelFunc
 
-	// ControllersStopped will be closed by the caller when the parser and
+	// ControllersStopped will be closed by the caller when the syncer and
 	// remediator have fully stopped. This unblocks Finalize() to destroy
 	// managed resource objects.
-	ControllersStopped <-chan struct{}
+	ControllersStopped []<-chan struct{}
```

Comment:
Dont love this interface. Slices are mutable. I wonder if it would make sense to use a waitgroup or a waitgroup blocking a goroutine that closes a single channel.